dashboard/app/dsl/script_dsl.rb

```diff
@@ -76,21 +78,43 @@ def level_concept_difficulty(json)
 
   string :video_key_for_next_level
 
+  boolean :active
+
   def assessment(name)
     level(name, {assessment: true})
   end
 
   def level(name, properties = {})
-    @levels << {
+    level = {
       :name => name,
-      :stage => @stage,
       :stage_flex_category => @stage_flex_category,
       :skin => @skin,
       :concepts => @concepts.join(','),
       :level_concept_difficulty => @level_concept_difficulty || {},
       :video_key => @video_key_for_next_level
     }.merge(properties).select{|_, v| v.present? }
     @video_key_for_next_level = nil
+    if @current_scriptlevel
+      @current_scriptlevel[:levels] << level
+      if !@active
+        @current_scriptlevel[:properties][name] = { :active => @active }
```

Comment:
You might want to take a look at the [`SerializedProperties`](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/concerns/serialized_properties.rb) concern, which is what we currently use to apply this pattern to the [`Level`](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/level.rb#L48-L57) and [`User`](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/user.rb#L78-L79) models. I'd recommend reusing this Concern if possible since it reduces the amount of (de)serialization code scattered through the implementation.